hazelcast/src/test/java/com/hazelcast/config/YamlConfigWithSystemPropertyTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.net.URLDecoder;
+
+import static java.io.File.createTempFile;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * These tests manipulate system properties, therefore they must be run in serial mode.
+ *
+ * @see XMLConfigWithSystemPropertyTest
+ */
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class YamlConfigWithSystemPropertyTest {
+
+    @Before
+    @After
+    public void beforeAndAfter() {
+        System.clearProperty("hazelcast.config");
+    }
+
+    @Test
+    public void testConfigurationWithFile() throws Exception {
+        URL url = getClass().getClassLoader().getResource("hazelcast-default.yaml");
+        assertNotNull(url);
```

Comment:
Again, I'd avoid having assertions in the middle of the tests.

Reply:
Removed as unnecessary. From the XML counterpart as well.